spec/services/group_suite_selector_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe GroupSuiteSelector do
+  describe '#select' do
+    context 'failure' do
+      it 'does not send emails' do
+        draw = instance_spy('Draw', next_groups: [])
+        group = instance_spy('group', suite: nil, draw: draw)
+        suite = mock_suite(id: 123, present: false)
+        described_class.select(group: group, suite_id: suite.id.to_s)
+        expect(group.draw).not_to have_received(:notify_next_groups)
+      end
+      it 'sets an error message in the flash' do
+        draw = instance_spy('Draw', next_groups: [])
+        group = instance_spy('group', suite: nil, draw: draw)
+        suite = mock_suite(id: 123, present: false)
+        result = described_class.select(group: group, suite_id: suite.id.to_s)
+        expect(result[:msg].keys).to include(:error)
+      end
+    end
+    context' success' do
+      # rubocop:disable RSpec/ExampleLength
+      it 'sends emails to the next groups' do
+        next_group = instance_spy('Group', lottery_number: 12)
```

Comment:
Can we be consistent about the capitalization of class names as passed to `instance_spy`? I think we should stick with camel case since that fixed that weird test before, but either way we should be consistent within a given test.